worlds/sm/__init__.py

```diff
@@ -285,28 +285,29 @@ def APPostPatchRom(self, romPatcher):
                                               "data", "SMBasepatch_prebuilt", "sm-basepatch-symbols.json"))
 
         # gather all player ids and names relevant to this rom, then write player name and player id data tables
-        self.playerIdSet: Set[int] = {0} # 0 is for "Archipelago" server
+        playerIdSet: Set[int] = {0}  # 0 is for "Archipelago" server
         for itemLoc in self.world.get_locations():
+            assert itemLoc.item
             # add each playerid who has a location containing an item to send to us *or* to an item_link we're part of
             if itemLoc.item.player == self.player or \
-                    (itemLoc.item.player in self.world.groups and \
+                    (itemLoc.item.player in self.world.groups and
                      self.player in self.world.groups[itemLoc.item.player]['players']):
-                self.playerIdSet |= {itemLoc.player}
+                playerIdSet |= {itemLoc.player}
             # add each playerid, including item link ids, that we'll be sending items to
             if itemLoc.player == self.player:
-                self.playerIdSet |= {itemLoc.item.player}
-        if len(self.playerIdSet) > SM_ROM_PLAYERDATA_COUNT:
+                playerIdSet |= {itemLoc.item.player}
+        if len(playerIdSet) > SM_ROM_PLAYERDATA_COUNT:
             # max 202 entries, but it's possible for item links to add enough replacement items for us, that are placed
             # in worlds that otherwise have no relation to us, that the 2*location count limit is exceeded
-            logger.warning(f"SM is interacting with too many players to fit in ROM. " +
-                            "Removing the highest {len(self.playerIdSet) - SM_ROM_PLAYERDATA_COUNT} ids to fit")
-            self.playerIdSet = set(sorted(self.playerIdSet)[:SM_ROM_PLAYERDATA_COUNT])
-        self.otherPlayerIndex : Dict[int, int] = {} # ap player id -> rom-local player index
-        playerNameData = []
-        playerIdData = []
+            logger.warning("SM is interacting with too many players to fit in ROM. "
+                           f"Removing the highest {len(playerIdSet) - SM_ROM_PLAYERDATA_COUNT} ids to fit")
+            playerIdSet = set(sorted(playerIdSet)[:SM_ROM_PLAYERDATA_COUNT])
+        otherPlayerIndex: Dict[int, int] = {}  # ap player id -> rom-local player index
+        playerNameData: List[Dict[str, Any]] = []
```

Comment:
The typing should apply to all or none of the constituent variables of patchDict further down:
```python
        patchDict = {   'MultiWorldLocations': multiWorldLocations,
                        'MultiWorldItems': multiWorldItems,
                        'offworldSprites': offworldSprites,
                        'deathLink': deathLink,
                        'remoteItem': remoteItem,
                        'ownPlayerId': ownPlayerId,
                        'playerNameData':  playerNameData,
                        'playerIdData':  playerIdData}
```

Reply:
I thought a `TypedDict` might be appropriate, but wasn't sure if it was worth it.
Also, Python 3.8 doesn't have some of the later features that make `TypedDict` easier to work with.

Reply:
`TypedDict` may be worth doing after the move to Python 3.10 then. Do you want to add the same typing to the other 6 local variables, or should I?

Reply:
I'll work on that.